Mage.Tests/src/test/java/org/mage/test/cards/conditional/SentinelSarahLyonsTest.java

```diff
@@ -0,0 +1,91 @@
+package org.mage.test.cards.conditional;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+public class SentinelSarahLyonsTest extends CardTestPlayerBase {
+
+    /*
+     * Sentinel Sarah Lyons {3}{R}{W}
+     * Legendary Creature - Human Knight (4/4)
+     * Haste
+     * As long as an artifact entered the battlefield under your control this turn, creatures you control get +2/+2.
+     * Battalion - Whenever Sentinel Sarah Lyons and at least two other creatures attack,
+     * Sentinel Sarah Lyons deals damage equal to the number of artifacts you control to target player.
+     */
+
+    @Test
+    public void test_ArtifactEntersUnderYourControlBeforeAndAfter() {
+        addCard(Zone.BATTLEFIELD, playerA, "Sentinel Sarah Lyons", 1);
+        addCard(Zone.HAND, playerA, "Darksteel Relic", 1); // Artifact w/ Indestructible
+
+        // Before casting an artifact (4/4)
+        setStrictChooseMode(true);
+        setStopAt(1, PhaseStep.UPKEEP);
+        execute();
+
+        assertPowerToughness(playerA, "Sentinel Sarah Lyons", 4, 4);
+
+        // Casting the artifact (should become a 6/6 until end of turn)
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Darksteel Relic");
+
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+
+        assertPowerToughness(playerA, "Sentinel Sarah Lyons", 6, 6);
+
+        // Turn after casting the artifact (should return to 4/4)
+        setStopAt(2, PhaseStep.UNTAP);
+        execute();
+
+        assertPowerToughness(playerA, "Sentinel Sarah Lyons", 4, 4);
+    }
+
+    @Test
+    public void test_ArtifactEntersUnderOpponentsControl() {
+        addCard(Zone.BATTLEFIELD, playerA, "Sentinel Sarah Lyons", 1);
+
+        addCard(Zone.HAND, playerB, "Darksteel Relic", 1);
+
+        castSpell(2, PhaseStep.PRECOMBAT_MAIN, playerB, "Darksteel Relic");
```

Comment:
Could you instead add a Vedalken Orrery to playerB, and cast the Relic turn 1?
Here the issue is that the Conditional/Watcher are not even used, since this is not playerA's turn, so it is less code coverage.